### ISSUE TITLE
@default not allowed by no-redundant-jsdoc even in non-functions

### ISSUE THREAD

**Issue by Jessidhia** [palantir/tslint#3419]
### Bug Report

- __TSLint version__: 5.8.0
- __TypeScript version__: 2.6.0
- __Running TSLint via__: CLI

#### TypeScript code being linted

```ts
interface Props {
  /** @default "bar" */
  foo?: string
}
```

with `tslint.json` configuration:

```json
{
  "rules": {
    "no-redundant-jsdoc": true
  }
}
```

#### Actual behavior

`JSDoc tag '@default' is redundant in TypeScript code. (no-redundant-jsdoc)`

#### Expected behavior

The `@default` tag is only actually redundant in function/method declarations, as you can write the argument defaults there. But it's not redundant in, for example, an interface declaration; as that default comes from another place (in this case, as it is a React component, from its `static defaultProps` property).

**Comment by JoshuaKGoldberg** [comment]
### 💀 _**It's time!**_ 💀

**TSLint is deprecated** and no longer accepting pull requests other than security fixes. See #4534. ☠️ 
We recommend you instead use **[typescript-eslint](https://typescript-eslint.io)** to lint your TypeScript code with **ESLint**. ✅

👋 It was a pleasure open sourcing with you!

**Comment by JoshuaKGoldberg** [comment]
🤖 Beep boop! 👉  **TSLint is deprecated** 👈 _(#4534)_  and you should **switch to [typescript-eslint](https://typescript-eslint.io)**! 🤖 

🔒 This issue is being locked to prevent further unnecessary discussions. Thank you! 👋
